test/stdlib/NSKeyedArchival.swift

```diff
@@ -0,0 +1,159 @@
+    var number: Int = 5
+    var string: String = "hello"
+    var data: Data = Data(bytes: [0, 1, 2, 3])
+    var array: [String] = ["stay", "a while", "and listen"]
+    var dictionary: [String:String] = ["Deckard": "Cain"]
+    static func ==(lhs: Simple, rhs: Simple) -> Bool {
+        return lhs.number == rhs.number && lhs.string == rhs.string && lhs.data == rhs.data && lhs.array == rhs.array && lhs.dictionary == rhs.dictionary
+    }
+}
+
+// simple struct comprised of entirely plist types
+struct NonPlist : Codable {
+    var date = Date()
+}
+
+// MARK - Tests
+@available(OSX 10.11, iOS 9.0, *)
+func test_codable_support_simple() {
+  let s = Simple()
+  
+  var data: Data? = nil
+  expectNoError {
+    data = try archive { archiver in
+      try archiver.encodeEncodable(s, forKey: "valid")
+      try archiver.encodeEncodable(s, forKey: "wrong-type")
+    }
+  }
+  
+  unarchive(data!) { unarchiver in
+    // confirm we can roundtrip our data
+    let roundtrip = unarchiver.decodeDecodable(Simple.self, forKey: "valid")
+    expectNotNil(roundtrip)
+    if let rt = roundtrip {
+        expectEqual(rt, s)
+    } 
+  
+    // also ask for something that is not there
+    let notThere = unarchiver.decodeDecodable(Simple.self, forKey: "not-there")
+    expectNil(notThere)
+    expectNil(unarchiver.error)
+  
+    // String != Simple so this should fail at the type level
+    let wrongType = unarchiver.decodeDecodable(String.self, forKey: "wrong-type")
+    expectNil(wrongType)
+    expectNotNil(unarchiver.error)
+  }
+}
+
+@available(OSX 10.11, iOS 9.0, *)
+func test_codeable_support_nonplist() {
+  expectError {
+    _ = try archive { archiver in
+      try archiver.encodeEncodable(NonPlist(), forKey: "non-codable")
+    }
+  }
+}
+
+@available(OSX 10.11, iOS 9.0, *)
+func test_codeable_support_noncodable() {
+  var data: Data? = nil
+  expectNoError {
+    data = archive { archiver in
+      archiver.encode(NSDate(), forKey: "non-codable")
+    }
+  } 
+
+  unarchive(data!) { unarchiver in
+    let nonCodable = unarchiver.decodeDecodable(Simple.self, forKey: "non-codable")
+    expectNil(nonCodable)
+    expectNotNil(unarchiver.error)
+  }
+}
+
+@available(OSX 10.11, iOS 9.0, *)
+func test_codeable_support_toplevel_noncodable() {
+  let s = Simple()
+  var data: Data? = nil
+  expectNoError {
+    data = try archive { archiver in
+      try archiver.encodeEncodable(s, forKey: "valid")
+      try archiver.encodeEncodable(Date(), forKey: "non-codable")
+    }
+  }
+
+  unarchive(data!) { unarchiver in
+    var caught = false
+    do {
+      _ = try unarchiver.decodeTopLevelDecodable(Simple.self, forKey: "non-codable")
+    } catch {
+      caught = true
+    }
+    expectTrue(caught)
+    
+    expectNoError {
+      let roundtrip = try unarchiver.decodeTopLevelDecodable(Simple.self, forKey: "valid")
+      expectNotNil(roundtrip)
+      if let rt = roundtrip {
+        expectEqual(rt, s)
+      }
+    }
+  }
+}
+
```

Comment:
Are these tests running successfully? Take a look at files like `CodableTests.swift` — we have a test harness at the bottom that forces the tests to run where `XCTest` isn't present. I hope these aren't passing by virtue of not running in the first place?